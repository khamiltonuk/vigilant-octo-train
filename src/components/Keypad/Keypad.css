.keypad {
  margin-bottom: 25px;
}
.operation-key,
.number-key {
  height: 60px;
  width: 20%;
  color: white;
  float: left;
}
.operation-key {
  background: #cf402c;
}
.operation-key:hover {
  background: #a13222;
}
.number-key {
  background: #744620;
}
.number-key:hover {
  background: #593619;
}
