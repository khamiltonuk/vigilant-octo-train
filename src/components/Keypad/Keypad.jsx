import React from "react";

import "./Keypad.css";

export default function Keypad(props) {
  const {
    keypadPress,
    clearInput,
    clearAll,
    addAll,
    sqaureAll,
    meanAll,
    getVariance,
    meanOfAllSquared
  } = props;
  return (
    <div className="keypad">
      <button className="number-key" onClick={() => keypadPress(0)}>
        0
      </button>
      <button className="number-key" onClick={() => keypadPress(1)}>
        1
      </button>
      <button className="number-key" onClick={() => keypadPress(2)}>
        2
      </button>
      <button className="number-key" onClick={() => keypadPress(3)}>
        3
      </button>
      <button className="number-key" onClick={() => keypadPress(4)}>
        4
      </button>
      <button className="number-key" onClick={() => keypadPress(5)}>
        5
      </button>
      <button className="number-key" onClick={() => keypadPress(6)}>
        6
      </button>
      <button className="number-key" onClick={() => keypadPress(7)}>
        7
      </button>
      <button className="number-key" onClick={() => keypadPress(8)}>
        8
      </button>
      <button className="number-key" onClick={() => keypadPress(9)}>
        9
      </button>
      <button className="operation-key" onClick={() => clearInput()}>
        clear
      </button>
      <button className="operation-key" onClick={() => clearAll()}>
        clear all
      </button>
      <button className="operation-key" onClick={() => addAll()}>
        add
      </button>
      <button className="operation-key" onClick={() => sqaureAll()}>
        square all
      </button>
      <button className="operation-key" onClick={() => meanAll()}>
        mean
      </button>
      <button className="operation-key" onClick={() => meanOfAllSquared()}>
        mean of squared
      </button>
      <button className="operation-key" onClick={() => getVariance()}>
        get variance
      </button>
    </div>
  );
}
