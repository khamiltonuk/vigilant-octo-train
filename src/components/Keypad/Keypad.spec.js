import React from "react";
import { shallow, simulate } from "enzyme";
import Keypad from "./Keypad";
import toJson from "enzyme-to-json";

const props = {
  keypadPress: () => true,
  clearInput: () => true,
  clearAll: () => true,
  calculate: () => true
};

describe("Keypad", () => {
  test("should return a 17 buttons", () => {
    const wrapper = shallow(<Keypad {...props} />);
    expect(wrapper.find("button").length).toBe(17);
  });

  test("should match the snapshot", () => {
    const wrapper = shallow(<Keypad {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test.skip("should behave...", () => {
    const props = {
      removeValue: () => true
    };
    const wrapper = shallow(<Keypad {...props} />);
    const button = wrapper.find("button").first();
    button.simulate("click");
    expect(wrapper.find("li").length).toBe(3);
  });
});
