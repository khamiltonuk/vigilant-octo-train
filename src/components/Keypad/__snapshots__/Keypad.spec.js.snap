// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keypad should match the snapshot 1`] = `
<div
  className="keypad group"
>
  <button
    className="number-key"
    key="0"
    onClick={[Function]}
  >
    0
  </button>
  <button
    className="number-key"
    key="1"
    onClick={[Function]}
  >
    1
  </button>
  <button
    className="number-key"
    key="2"
    onClick={[Function]}
  >
    2
  </button>
  <button
    className="number-key"
    key="3"
    onClick={[Function]}
  >
    3
  </button>
  <button
    className="number-key"
    key="4"
    onClick={[Function]}
  >
    4
  </button>
  <button
    className="number-key"
    key="5"
    onClick={[Function]}
  >
    5
  </button>
  <button
    className="number-key"
    key="6"
    onClick={[Function]}
  >
    6
  </button>
  <button
    className="number-key"
    key="7"
    onClick={[Function]}
  >
    7
  </button>
  <button
    className="number-key"
    key="8"
    onClick={[Function]}
  >
    8
  </button>
  <button
    className="number-key"
    key="9"
    onClick={[Function]}
  >
    9
  </button>
  <button
    className="operation-key"
    onClick={[Function]}
  >
    clear
  </button>
  <button
    className="operation-key"
    onClick={[Function]}
  >
    clear all
  </button>
  <button
    className="operation-key"
    key="0"
    onClick={[Function]}
  >
    Add All
  </button>
  <button
    className="operation-key"
    key="1"
    onClick={[Function]}
  >
    Aquare All
  </button>
  <button
    className="operation-key"
    key="2"
    onClick={[Function]}
  >
    Mean All
  </button>
  <button
    className="operation-key"
    key="3"
    onClick={[Function]}
  >
    Mean of All Squared
  </button>
  <button
    className="operation-key"
    key="4"
    onClick={[Function]}
  >
    Get Variance
  </button>
</div>
`;
